pinot-query-planner/src/main/java/org/apache/pinot/query/catalog/PinotCatalog.java

```diff
@@ -80,8 +80,9 @@ public Table getTable(String name) {
    */
   @Override
   public Set<String> getTableNames() {
-    return _tableCache.getTableNameMap().keySet().stream().filter(n -> DatabaseUtils.isPartOfDatabase(n, _databaseName))
-        .collect(Collectors.toSet());
+    //return _tableCache.getTableNameMap().keySet().stream().filter(n -> DatabaseUtils.isPartOfDatabase(n, _databaseName))
+    //    .collect(Collectors.toSet());
+    return _tableCache.getTableNameMap().keySet();
```

Comment:
Enable's `calcite` case-insensitive configuration led to a lot of unexpected consequences. One of them is that, during table name resolution and validation, it now relies in `schema.getTableNames()` to find if a table exists:

https://github.com/apache/calcite/blob/f9cc84f35aeba0effd6654da8471d4bef9eaa925/core/src/main/java/org/apache/calcite/jdbc/SimpleCalciteSchema.java#L122

As it was implemented, this method got rid of all non-prefixed tableNames when `_databaseName != null`, and probably that was not intended.

Reply:
> As it was implemented, this method got rid of all non-prefixed tableNames when _databaseName != null, and probably that was not intended.

The non-prefixed table names are assumed to belong to the "default" database IIUC. So shouldn't we want to exclude them when there's a non-default database being used?

cc - @shounakmk219

Reply:
That's an interesting thought.

I guess the proper response is to return only tables that belong to the implicit database, but strip them from the database prefix (or, at least, return them both in prefixed and non-prefixed form). For instance `SimpleCalciteSchema` will lookup for the table name without its prefix, so we cannot expect only one of the two forms.

https://github.com/apache/pinot/pull/14830/commits/97870597421d0c1cbb50634d1913701375e163c3